mautic/mautic
# update list of default themes to be in line with the current ones

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

The current list of default themes is outdated (  = it does not correspond with the actual themes in the `themes` folder)
This leads to 2 issues:

- Some default themes can be deleted
- A custom theme named `system` can never be deleted, as it's name is still in the list of default themes





#### Steps to test this PR:

1. set up a vanilla mautic, 
1. go to the `themes` page
1. check you can't delete the `blank` theme (OK), but can delete the `trulypersonal` theme (not OK)
1. apply the patch
1. reset / reinstall the vanilla mautic
1. refresh the `themes` page
1. check you can't delete the `blank` theme (OK) and the `trulypersonal` theme (OK)





## other remarks

- There is actually 1 theme named `Mauve` in the monorepo, but named `mauve` in the separate repo.
This difference in casing may lead to strange situations, so I'm not sure why it's capitalised in the monorepo
- the hardcoded `$defaultThemes` property could become completely obsolete when themes uploaded by users are no longer stored in the `themes` folder, but e.g. in `media/themes`


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10731"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
